common/web/types/src/ldml-keyboard/pattern-parser.ts

```diff
@@ -197,13 +188,14 @@ export class MarkerParser {
       const rest = a.slice(i).join('');
       const p = MarkerParser.parse_next_marker(rest, forMatch);
       const have_marker = !!(p?.match);
+      const has_nfd_boundary_before = (nfdNoBoundaryBefore.indexOf(a[i]?.codePointAt(0)) === -1);
```

Comment:
When can `a[i]` be undefined and not a string? If `a[i]` is guaranteed to be a string then we should not be using optional chaining.

```suggestion
      const has_nfd_boundary_before = (nfdNoBoundaryBefore.indexOf(a[i].codePointAt(0)) === -1);
```

Reply:
when `i === str_end` (see line 203) - this loop is called once at the end of the string. 

so `a[i]` can be undefined.

Reply:
can you add a comment accordingly somewhere? Because that's not necessarily intuitive to run loop past the end :grin: